ignite/metrics/fid.py

```diff
@@ -0,0 +1,131 @@
+import os
+from typing import Callable, Sequence, Union
+
+import torch
+from scipy import linalg
+from torchvision import transforms
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+__all__ = ["FID"]
+
+class DefaultDataset(data.Dataset):
```

Comment:
I think we should not deal with datasets, but have rely on batches provided by the evaluator.